File: Python/Pt1Exercise5/harmonic.py
"""
 Exercise 5 from the Python tutorial Part 1 on:
 https://codeandwork.github.io/courses/prep/pythonTutorial1.html
"""

sum = 1
for n in range(1, 6):
    if n == 1:
        print(sum)
    if n > 1:
        step = 1/n
        sum += step
        print(sum)

File: Java/Pt1Exercise5/Harmonic.java
/*
 * Exercise 5 from the Python tutorial Part 1 on:
 * https://codeandwork.github.io/courses/prep/pythonTutorial1.html

 * Exercise must be written in Java.
 */

public class Harmonic {

    public static void main(String args[]) {
        double sum = 1.0;
        for (int i = 1; i <= 5; i++) {
            // Print the first term of the harmonic sequence.
            if (i == 1) {
                System.out.println((int) sum);
            }
            // Compute and print the remaining four terms of the harmonic sequence.
            if (i > 1) {
                double step = (double) 1 / i;
                sum = sum + step;
                System.out.println(sum);
            }
        }
    }
}
